orchestrator/src/fund.go

```diff
@@ -140,4 +142,34 @@ func (FundAction) RunMany(config Config, actionIOs []ActionIO) error {
 	return nil
 }
 
+func (FundAction) Validate(rawParams json.RawMessage) error {
+	var params FundParams
+	if err := json.Unmarshal(rawParams, &params); err != nil {
+		return fmt.Errorf("failed to unmarshal fund-keys params: %v", err)
+	}
+	fundKeysBaseDir := extractBaseDir(params.Prefix)
+	if folderExists(fundKeysBaseDir) {
+		return fmt.Errorf("directory '%s' already exists. Please re-generate script using unique experiment name or different '-fund-keys-dir' CLI argument value", fundKeysBaseDir)
+	}
+	return nil
+}
+
+// Helper function to extract the first two levels of the path
+func extractBaseDir(prefix string) string {
+	parts := strings.Split(filepath.ToSlash(prefix), "/")
```

Comment:
Thanks @georgeee.
The Generator seems to generate prefix for `fund-keys` action in a form of:

- `./fund-keys/<experiment_name>/round-<N>/payments/key`
- `./fund-keys/<experiment_name>/round-<N>/zkapps/key`

If use r will specify the `-fund-keys-dir`CLI argument then the only difference will be in replaced `./fund-keys` with whatever will be provided by user.
Hence, it seems to be safe to simply count slashed from the end, isn't it?

Reply:
If user specifies `./my-fund-keys/some/long/deep/directory/hidden/for/whatever/reason`, code here would return `./my-fund-keys/some` instead of `./my-fund-keys/some/long/deep/directory/hidden/for/whatever/reason/<experiment_name>`.

At least this was my understanding of the code

Reply:
https://github.com/o1-labs/mina-perf-testing/pull/2/commits/7e37342b89f1367d6902cf1244907d19e5db4876